tools/testsocket/testsocket.cpp

```diff
@@ -733,9 +741,12 @@ int doSendQuery(const char * ip, unsigned port, const char * base)
                 fprintf(trace, "%s", result.str());
             }
 
+            double queryTimeMS = (double)(cycle_to_nanosec(endtime - starttime)/1000000);
```

Comment:
minor: I think you meant to cast before the divide otherwise you lose some accuracy.

Reply:
@ghalliday thanks, good find.  Yes I copied/moved the existing code without checking it, sorry.
I fixed it here and in one other place below.  If you are ok with it I'll squash.